qui/tray/devices.py

```diff
@@ -329,12 +329,11 @@ def on_label_changed(self, vm, _event, **_kwargs):
             name = vm.name
         except qubesadmin.exc.QubesPropertyAccessError:
             return  # the VM was deleted before its status could be updated
-
         for domain in self.vms:
-            if domain.name == name:
+            if str(domain) == name:
                 domain.icon = vm.label.icon
 
-        for device in self.devices:
+        for device in self.devices.items():
```

Comment:
This should be `values()`, not `items()`. `items()` returns tuples (key, value), not just values of a dict.